### PR TITLE
GC-9120 nodejs update

### PR DESCRIPTION
**Pull request by shawnjoseph**
Nodejs 8.X is being deprecated, as a result the lambda needs to be updated to nodejs 10+. In this case we updated to nodeJS 12.X (done in the lambda configs) since changes were minimal.

* Mainly just cleaning up some code: using `let` and `const` instead of `var`, making the cloudwatch logger a bit cleaner, getting rid of redundant variables.

### CODE REVIEW

**Comment by ramonPaulAlvarez** [comment]
I think `.idea/vcs.xml` may need to be removed.

File: .gitignore [hunk 1]
```diff
@@ -61,4 +61,10 @@ typings/
 .serverless
 
 # Package-lock
-package-lock.json
\ No newline at end of file
+package-lock.json
+
+# Pycharm
+.idea
```
[the remark this answers is not in the document]
Comment (shawnjoseph) [reply .gitignore]: ah, good idea! 👏

[the remark this answers is not in the document]
Comment (shawnjoseph) [reply .gitignore]: a good idea! 👏

File: lambda_function.js [hunk 2]
```diff
@@ -568,7 +567,7 @@ async function handleSync(body, token) {
                 let device_tmpl = await JSON.parse(JSON.stringify(DEVICE_TEMPLATE));
                 device_tmpl.id = resp[i].id;
                 device_tmpl.name.name = resp[i].name;
-                online = await resp[i].meta.find(x => {
+                let online = await resp[i].meta.find(x => {
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
                const online = await resp[i].meta.find(x => {
```

File: lambda_function.js [hunk 3]
```diff
@@ -223,7 +223,7 @@ function combineCommands(action, value, successfulDevices, failedDevices) {
             break;
         case "brightness":
             if (successfulDevices) {
-                var cmd = {
+                let cmd = {
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
                const cmd = {
```

File: lambda_function.js [hunk 4]
```diff
@@ -233,7 +233,7 @@ function combineCommands(action, value, successfulDevices, failedDevices) {
                 payloadCommands.push(cmd);
             }
             if (failedDevices) {
-                var cmd = {
+                let cmd = {
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
                const cmd = {
```

File: lambda_function.js [hunk 5]
```diff
@@ -214,7 +214,7 @@ function combineCommands(action, value, successfulDevices, failedDevices) {
                 payloadCommands.push(cmd);
             }
             if (!Array.isArray(failedDevices) || failedDevices.length) {
-                var cmd = {
+                let cmd = {
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
                const cmd = {
```

File: lambda_function.js [hunk 6]
```diff
@@ -203,7 +203,7 @@ function combineCommands(action, value, successfulDevices, failedDevices) {
     switch (action) {
         case "on":
             if (successfulDevices) {
-                var cmd = {
+                let cmd = {
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
                const cmd = {
```

File: lambda_function.js [hunk 7]
```diff
@@ -142,7 +142,7 @@ async function getRequest(endpoint, token, is_next = false) {
     return await rp(options)
         .then(async function(r) {
             // Provide the response body of this page and the next.
-            var links = r.headers.link ? parse(r.headers.link) : null;
+            let links = r.headers.link ? parse(r.headers.link) : null;
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
            const links = r.headers.link ? parse(r.headers.link) : null;
```

File: lambda_function.js [hunk 8]
```diff
@@ -88,7 +89,7 @@ async function getStates(token) {
             return deviceStates;
         })
         .then(async function(deviceStates) {
-            spaceStates = await helpers
+            let spaceStates = await helpers
```
Comment (dustinrohde) [inline lambda_function.js]:
```suggestion
            const spaceStates = await helpers
```

**Comment by dustinrohde** [comment]
looks good, just recommending that `const` be used over `let` unless the variable is being modified

File: lambda_function.js [hunk 9]
```diff
@@ -629,73 +627,76 @@ async function handleSync(body, token) {
...
+    } catch (e) {
+        logger.warn("Service account key is not found. Report state will be unavailable.");
+    }
+
+    global.app = smarthome({
+        jwt: homegraph,
+    });
+
+    try {
+        let body;
```
[the remark this answers is not in the document]
Comment (shawnjoseph) [reply lambda_function.js]: I think due to the nature of `let` if i add them in the `if` statement below, I won't be able to access their proper value outside the `if`

**Pull request merged**